### PR TITLE
[Dashboard] feat: added button to view deployed contract on block explorer

### PR DESCRIPTION
**Pull request by ThianHooi**
- [x] feat: added a button on the [deployed contract page](https://thirdweb.com/dashboard/contracts) which opens a new tab to view the deployed contract on the block explorer

The outcome is as such:
- Notice the icon beside the contract address
image

### CODE REVIEW

File: components/contract-components/tables/deployed-contracts.tsx [hunk 1]
```diff
@@ -393,7 +395,25 @@ export const ContractTable: ComponentWithChildren<ContractTableProps> = ({
...
+            configuredChains[cell.row.original.chainId] ||
+            chainIdToChainRecord[cell.row.original.chainId];
+          const blockExplorerUrl = data?.explorers?.[0]?.url;
+
+          return (
+            <Flex align="center" gap={2}>
+              <AddressCopyButton address={cell.row.original.address} />
+              {blockExplorerUrl && (
+                <Link
+                  href={`${blockExplorerUrl}/address/${cell.row.original.address}`}
```
Comment (jnsdls) [inline components/contract-components/tables/deployed-contracts.tsx:408]: is this guaranteed to be the correct URL scheme for all variants of block explorers?

Comment (ThianHooi) [reply components/contract-components/tables/deployed-contracts.tsx:411]: Agree. I added new changes to filter block explorer URL that follows the standard of [EIP3091](https://eips.ethereum.org/EIPS/eip-3091)

Comment (ThianHooi) [reply components/contract-components/tables/deployed-contracts.tsx:411]: > is this guaranteed to be the correct URL scheme for all variants of block explorers?

Agree. I added new changes to filter block explorer URL that follows the standard of [EIP3091](https://eips.ethereum.org/EIPS/eip-3091)

**Comment by ThianHooi** [comment]
> hey @ThianHooi thanks for this contribution!
> 
> Out of interest what functionality is missing from the dashboard itself that you would need to go to a blockexplorer for in the first place?
> 
> While this seems useful at first glance ultimately we'd want to cover any such functionality (or at least most of it) directly within the dashboard. :)

@jnsdls thirdweb dashboard does cover a lot of functionality, especially for prebuilt contracts. However, functionalities such as viewing list of transactions, viewing balance of tokens of a contract, and viewing the analytics, can be done only on block explorer

**Comment by jnsdls** [comment]
> > hey @ThianHooi thanks for this contribution!
> > Out of interest what functionality is missing from the dashboard itself that you would need to go to a blockexplorer for in the first place?
> > While this seems useful at first glance ultimately we'd want to cover any such functionality (or at least most of it) directly within the dashboard. :)
> 
> @jnsdls thirdweb dashboard does cover a lot of functionality, especially for prebuilt contracts. However, functionalities such as viewing list of transactions, viewing balance of tokens of a contract, and viewing the analytics, can be done only on block explorer

thx this is a good list of things for us to add to the dashboard capabilities (we're already working on some of the underlying requirements of these)

In the meantime I'm happy to add something like your PR. Instead of adding it into the table could you add it to the contract overview page itself (I think that would be a better place for it, you also get some more space to maybe add the name of the explorer into the link?)

**Comment by ThianHooi** [comment]
- [x] feat: added a button on the contract page which opens a new tab to view the deployed contract on the block explorer

Outcome as such:

image

**Comment by ThianHooi** [comment]
@jnsdls Thanks for merging the PR.

I actually have been trying to reach out to you on Twitter. I really hope you would spend some time to go through your message request and read my DM. (So sorry to reach out to you using this way on Github)

**Pull request merged**